plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePluginImpl.kt

```diff
@@ -0,0 +1,72 @@
+package com.mapbox.maps.plugin.lifecycle
+
+import android.content.ComponentCallbacks
+import android.content.res.Configuration
+import android.widget.FrameLayout
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import com.mapbox.common.Logger
+import com.mapbox.maps.MapboxLifecycleObserver
+import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
+
+/**
+ * Concrete implementation of MapboxLifecyclePlugin.
+ */
+class MapboxLifecyclePluginImpl : MapboxLifecyclePlugin {
+  /**
+   * Register a MapboxLifecycleObserver to observe life cycle events from LifecycleOwner
+   *
+   * @param mapView the instance of mapView, will get the LifecycleOwner from mapview's parent
+   * @param observer the observer that listen to the life cycle events
+   */
+  override fun registerLifecycleObserver(mapView: FrameLayout, observer: MapboxLifecycleObserver) {
+    val lifecycleOwner = ViewTreeLifecycleOwner.get(mapView)
+    if (lifecycleOwner == null) {
+      Logger.w(
+        TAG,
+        "Can't get lifecycleOwner for mapview, please make sure the host Activity is AppCompatActivity"
```

Comment:
Could this be an error message? and we should provide what to do in this case, if the host Activity is not AppCompatActivity, the user should follow the instructions to manually overwrite their activity/fragment's lifecycle events.